--- main.cpp.orig	2019-08-26 22:51:44.229281866 +0900
+++ main.cpp	2019-08-26 22:56:32.169706856 +0900
@@ -34,6 +34,8 @@
 **
 ****************************************************************************/
 
+#include <windows.h>
+#include <tchar.h>
 #include <SDL.h>
 #include <SDL_gamecontroller.h>
 
@@ -43,3 +45,12 @@
     SDL_Quit();
     return 0;
 }
+
+int WINAPI WinMain(
+		     HINSTANCE hCurInst,
+		     HINSTANCE hPrevInst,
+		     LPSTR lpsCmdLine,
+		     int nCmdShow)
+{
+  return  main(0, NULL);
+}
